public/apps/configuration/utils/password-edit-panel.tsx

```diff
@@ -13,57 +13,48 @@
  *   permissions and limitations under the License.
  */
 
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { EuiFieldText, EuiIcon } from '@elastic/eui';
 import { FormRow } from './form-row';
 
-// At least one uppercase, one lowercase, one digit and one special character
-// and at least 8 characters long
-const PASSWORD_PATTERN = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?([^\w\s]|[_])).{8,}$/;
-
-// TODO: Add unit test
-function isValidPassword(password: string) {
-  return PASSWORD_PATTERN.test(password);
-}
-
 export function PasswordEditPanel(props: {
   updatePassword: (p: string) => void;
   updateIsInvalid: (v: boolean) => void;
 }) {
   const [password, setPassword] = useState<string>('');
-  const [isPasswordInvalid, setIsPasswordInvalid] = useState<boolean>(false);
+  const [repeatPassword, setRepeatPassword] = useState<string>('');
   const [isRepeatPasswordInvalid, setIsRepeatPasswordInvalid] = useState<boolean>(false);
 
+  useEffect(() => {
+    props.updatePassword(password);
+    const isValid = repeatPassword !== password;
```

Comment:
This variable looks like isInvalid instead of isValid.

Reply:
Good catch, almost introduced another bug.